sites/default/statement.inc.php

```diff
@@ -21,13 +22,384 @@
+    if ($tmp == 'Procedure 9920' || $tmp == 'Procedure 9921' || $tmp == 'Procedure 9200' || $tmp == 'Procedure 9201')
+      $description = str_replace("Procedure",xl('Office Visit').":",$description);
+    //92002-14 are Eye Office Visit Codes
+
+    $dos = $line['dos'];
+    ksort($line['detail']);
+      # Compute the aging bucket index and accumulate into that bucket.
+    $age_in_days = (int) (($todays_time - strtotime($dos)) / (60 * 60 * 24));
+    $age_index = (int) (($age_in_days - 1) / 30);
+    $age_index = max(0, min($num_ages - 1, $age_index));
+    $aging[$age_index] += $line['amount'] - $line['paid'];
+
+    foreach ($line['detail'] as $dkey => $ddata) {
+      $ddate = substr($dkey, 0, 10);
+      if (preg_match('/^(\d\d\d\d)(\d\d)(\d\d)\s*$/', $ddate, $matches)) {
+        $ddate = $matches[1] . '-' . $matches[2] . '-' . $matches[3];
+      }
+      $amount = '';
+
+      if ($ddata['pmt']) {
+        $amount = sprintf("%.2f", 0 - $ddata['pmt']);
+        $desc = xl('Paid') .' '. oeFormatShortDate($ddate) .': '. $ddata['src'].' '. $ddata['pmt_method'].' '. $ddata['insurance_company'];
+      } else if ($ddata['rsn']) {
+        if ($ddata['chg']) {
+          $amount = sprintf("%.2f", $ddata['chg']);
+          $desc = xl('Adj') .' '.  oeFormatShortDate($ddate) .': ' . $ddata['rsn'].' '.$ddata['pmt_method'].' '. $ddata['insurance_company'];
+        } else {
+          $desc = xl('Note') .' '. oeFormatShortDate($ddate) .': '. $ddata['rsn'].' '.$ddata['pmt_method'].' '. $ddata['insurance_company'];
+        }
+      } else if ($ddata['chg'] < 0) {
+        $amount = sprintf("%.2f", $ddata['chg']);
+        $desc = xl('Patient Payment');
+      } else {
+        $amount = sprintf("%.2f", $ddata['chg']);
+        $desc = $description;
+
+      }
+
+      $out .= sprintf("%-10s  %-45s%8s\n", oeFormatShortDate($dos), $desc, $amount);
+      $dos = '';
+      ++$count;
+    }
+  }
+
+  // This generates blank lines until we are at line 20.
+  //  At line 20 we start middle third.
+
+  while ($count++ < 16) $out .= "\n";
+    # Generate the string of aging text.  This will look like:
+    # Current xxx.xx / 31-60 x.xx / 61-90 x.xx / Over-90 xxx.xx
+    # ....+....1....+....2....+....3....+....4....+....5....+....6....+
+    #
+  $ageline = xl('Current') .': ' . sprintf("%.2f", $aging[0]);
+  for ($age_index = 1; $age_index < ($num_ages - 1); ++$age_index) {
+    $ageline .= ' | ' . ($age_index * 30 + 1) . '-' . ($age_index * 30 + 30) . ':' . 
+                sprintf(" %.2f", $GLOBALS['gbl_currency_symbol'].''.$aging[$age_index]);
+  }
+
+  // Fixed text labels
+  $label_ptname = xl('Name');
+  $label_today = xl('Date');
+  $label_due = xl('Due');
+  $label_thanks = xl('Thank you for choosing');
+  $label_call = xl('Please call if any of the above information is incorrect.');
+  $label_prompt = xl('We appreciate prompt payment of balances due.');
+  $label_dept = xl('Billing Department');
+  $label_bill_phone = xl('(413) 525-1611');
```

Comment:
Should not be hard-coding a phone number unless your ready to get a lot of phone calls from other OpenEMR installs :)